### PR TITLE
Add check and quota types for IBP

### PR DESCRIPTION
**Pull request by guptasu**
Add some static types (CheckResult, QuotaResult, QuotaRequest) for IBP interface. This is slightly different than the current Go LAP for Quota. We do not ask for multiple quotas, but there is no reason to not ask for multiple quota at once from a backend/adapter.

### CODE REVIEW

File: mixer/adapter/model/v1beta1/check.proto [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import "google/protobuf/duration.proto";
+import "google/rpc/status.proto";
+
+option (gogoproto.goproto_getters_all) = false;
+option (gogoproto.equal_all) = false;
+option (gogoproto.gostring_all) = false;
+
+// // Expresses the result of a precondition check.
+message CheckResult {
+  // A status code of OK indicates preconditions was satisfied. Any other code indicates preconditions was not
```
Comment (geeknoid) [inline mixer/adapter/model/v1beta1/check.proto]: preconditions was -> were
x2

File: mixer/adapter/model/v1beta1/check.proto [hunk 2]
```diff
@@ -0,0 +1,38 @@
+// Copyright 2016 Istio Authors
```
Comment (geeknoid) [inline mixer/adapter/model/v1beta1/check.proto]: Invalid year, here and elsewhere

File: mixer/adapter/model/v1beta1/quota.proto [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+import "gogoproto/gogo.proto";
+import "google/protobuf/duration.proto";
+import "google/rpc/status.proto";
+
+option (gogoproto.goproto_getters_all) = false;
+option (gogoproto.equal_all) = false;
+option (gogoproto.gostring_all) = false;
+
+// Used to get a thumbs-up/thumbs-down for quota allocation before performing an action.
```
Comment (geeknoid) [inline mixer/adapter/model/v1beta1/quota.proto]: Not really for thumbs up/down. Better to just say "Used for quota allocation..."

File: mixer/adapter/model/v1beta1/quota.proto [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+  // The individual quotas to allocate
+  map<string, QuotaParams> quotas = 1 [(gogoproto.nullable) = false];
+}
+
+// Expresses the result of multiple quota allocations.
+message QuotaResult {
+  // Expresses the result of a quota allocation.
+  message Result {
+    // A status code of OK indicates quota allocation was successful. Any other code indicates quota allocation was not
```
Comment (geeknoid) [inline mixer/adapter/model/v1beta1/quota.proto]: I don't think we need this, we should get rid of the equivalent in the LAP too for that matter.

Comment (guptasu) [reply mixer/adapter/model/v1beta1/quota.proto]: D

File: mixer/adapter/model/v1beta1/quota.proto (hunk 3 above)
Comment (guptasu) [reply mixer/adapter/model/v1beta1/quota.proto]: D

File: mixer/adapter/model/v1beta1/check.proto (hunk 2 above)
Comment (guptasu) [reply mixer/adapter/model/v1beta1/check.proto]: D

File: mixer/adapter/model/v1beta1/check.proto (hunk 1 above)
Comment (guptasu) [reply mixer/adapter/model/v1beta1/check.proto]: D

File: mixer/adapter/model/v1beta1/quota.proto [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    // successful and details describe why.
+    google.rpc.Status status = 1 [(gogoproto.nullable) = false];
```
Comment (geeknoid) [inline mixer/adapter/model/v1beta1/quota.proto]: Still here...

File: mixer/adapter/model/v1beta1/report.proto [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+option (gogoproto.goproto_getters_all) = false;
+option (gogoproto.equal_all) = false;
+option (gogoproto.gostring_all) = false;
+
+// Expresses the result of a report call.
+message ReportResult {
+  // A status code of OK indicates report was successful. Any other code indicates report was not
+  // satisfied and details describe why.
+  google.rpc.Status status = 1 [(gogoproto.nullable) = false];
```
Comment (geeknoid) [inline mixer/adapter/model/v1beta1/report.proto]: No need for status.

**Comment by guptasu** [comment]
@geeknoid .. hmm. I thought I removed the status, but it seems I didn't. Weird.

**Comment by guptasu** [comment]
@geeknoid all good ?

**Pull request merged**